### ISSUE TITLE
Dependency on inherited test

### ISSUE THREAD

**Issue by JasperHorn** [sebastianbergmann/phpunit#945]
Imagine the following testing code:

    abstract class ParentTest extends PHPUnit_Framework_TestCase
    {
         public function testA() {..}
    }

    class ChildTest extends ParentTest
    {
         /**
          * @depends testA
          */
         public function testB() {..}
    }

The way things are now, inherited tests are run *after* tests defined in the Child class, meaning that testB is always skipped. As far as I could find, there is no way to make sure that `testA()` runs first without overriding it.

If instead, inherited tests are executed first, dependencies such as the one above would work. I believe this change in ordering should have no averse effects. 

Order shouldn't matter except for the dependencies given and it makes no sense to have a test from the parent depend on one from the child, so this change in order shouldn't change which tests fail and which do not. The only other side effect should be a change in order of the output, but I'd say this actually puts things in a better order. (Basically, if there are dependencies but they aren't declared to PHPUnit, the more "relevant" failure will be shown.)

In summary: I propose inherited tests are run before tests defined in the child class, so you can define dependencies between the two.

**Comment by sebastianbergmann** [comment]
Dear contributor,

let me start by apologizing for not commenting and/or working on the issue you have reported or merging the pull request you have sent sooner.

PHPUnit 5.0 was released today. And today I am closing all open bug reports and pull requests for PHPUnit and its dependencies that I maintain. Please do not interpret the closing of this ticket as an insult or a lack of interest in your problem. I am sorry for any inconvenience this may cause.

If the topic of this ticket is still relevant then please open a new ticket or send a new pull request. If your ticket or pull request is about a defect then please check whether the issue still exists in PHPUnit 4.8 (which will received bug fixes until August 2016). If your ticket or pull request is about a new feature then please port your patch PHPUnit 5.0 before sending a new pull request.

I hope that today's extreme backlog grooming will allow me to respond to bug reports and pull requests in a more timely manner in the future.

Thank you for your understanding,
Sebastian
